lib/ansible/modules/cloud/amazon/aws_step_functions_state_machine_execution.py

```diff
@@ -103,40 +103,60 @@
 
 
 def start_execution(module, sfn_client):
-    check_mode(module, msg='State machine execution would be started.', changed=True)
+    '''
+    start_execution uses execution name to determine if a previous execution already exists.
+    If an execution by the provided name exists, call client.start_execution will not be called.
+    '''
 
     state_machine_arn = module.params.get('state_machine_arn')
     name = module.params.get('name')
     execution_input = module.params.get('execution_input')
 
     try:
-        execution = sfn_client.start_execution(
+        # list_executions is eventually consistent
+        resp_iterators = sfn_client.get_paginator('list_executions').paginate(stateMachineArn=state_machine_arn)
+
+        for resp in resp_iterators:
```

Comment:
Paginators have a `build_full_result` method you can use, which is a little nicer than nested loops. So you could do:
```
current_executions = resp_iterators.build_full_result()['executions']
for execution in current_executions:
    if name == execution['name']:
```

Reply:
Thanks, I will make the required changes. Does it look good otherwise?